rust/template/differential_datalog/src/program/mod.rs

```diff
@@ -110,49 +110,62 @@ pub type ArrId = (RelId, usize);
 /// Function type used to map the content of a relation
 /// (see `XFormCollection::Map`).
 pub type MapFunc = fn(DDValue) -> DDValue;
+pub type MapDynFunc = Arc<dyn Fn(DDValue) -> DDValue + Sync + Send>;
```

Comment:
I actually think boxing and having the user take responsibility for synchronizing is better & lower overhead

Reply:
Changing from `Arc` to `Box` causes some troubles because `MapDynFunc` is used in the field of `XFormCollection` which requires `Clone` but you can't add `Clone` to trait object inside `MapDynFunc`.